blockchain/block/genesis_block.go

```diff
@@ -6,43 +6,24 @@
 package block
 
 import (
-	"sync"
 	"time"
 
 	"github.com/iotexproject/go-pkgs/hash"
 
-	"github.com/iotexproject/iotex-core/v2/blockchain/genesis"
 	"github.com/iotexproject/iotex-core/v2/pkg/version"
 )
 
-var (
-	_loadGenesisHash sync.Once
-	_genesisHash     hash.Hash256
-)
-
 // GenesisBlock returns the genesis block
-func GenesisBlock() *Block {
+func GenesisBlock(timestamp time.Time) *Block {
 	return &Block{
 		Header: Header{
 			version:          version.ProtocolVersion,
 			height:           0,
-			timestamp:        time.Unix(genesis.Timestamp(), 0),
+			timestamp:        timestamp,
```

Comment:
Does this solve the issue of `block.Hash()` not being equal to `genesis.Hash()` when the block height equals 0?

Reply:
sadly, no...
with current implementation, `block(0)` shouldn't return a block, but throw an error.
we made a mistake here: `block(0).hash() == genesis.Hash()`. if we were able to redesign the blockchain, `block(0).prevHash == genesis.Hash()` and `block(0)` is the very first block instead of a virtual block.